src/component/CreateBankAccount.js

```diff
@@ -0,0 +1,92 @@
+import React, { useRef, useState } from 'react'
+import { Form, Button, Card, Alert } from 'react-bootstrap'
+import { AiOutlineMail, AiOutlineUser } from 'react-icons/ai'
+import { RiLockPasswordLine } from 'react-icons/ri'
+import { FaAddressBook } from 'react-icons/fa'
+import { Link, useHistory } from 'react-router-dom'
+
+import { useAuth } from '../contexts/AuthContext'
+import BankAccount from '../models/BankAccount'
+
+export function CreateBankAccount() {
+  const accountNicknameRef = useRef()
+  const currencyRef = useRef()
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+  const { currentUser } = useAuth()
+
+  async function generateIBAN() {
+    while (1) {
+      let IBAN = "RO69DOGE"
+      for (let i = 0; i < 16; i++) {
+        IBAN += Math.floor(Math.random() * 10).toString()
```

Comment:
You could use 
```
IBAN += [...Array(16).keys()].map(() => Math.floor(Math.random() * 10)).join('')
```
to generate 16 characters instead of a for. It's cleaner.